src/groupeddataframe/grouping.jl

```diff
@@ -387,6 +423,98 @@ function gen_fun(::Type{NT}, f2) where NT
     end
 end
 
+# Generate a named tuple of reductions to call for each group
+function gen_fun(::Type{NT}, f2::Tuple{Vararg{Pair{<:Val, <:Reduce}}}) where NT
+    tup = map(f2) do p
+        nms = get_val(first(p))
+        nms => last(p)
+    end
+    NT(tup)
+end
+
+# Use reducedim_init to get a vector of the right type,
+# but trick it into using the expected length
+groupreduce_init(op, incol, gd) =
+    Base.reducedim_init(identity, op, view(incol, 1:length(gd)), 2)
+
+for (op, initf) in ((:max, :typemin), (:min, :typemax))
+    @eval begin
+        function groupreduce_init(::typeof($op), incol::AbstractVector{T}, gd) where T
+            outcol = similar(incol, length(gd))
+            # Comparison is possible only between CatValues from the same pool
```

Comment:
It's unfortunate that the standard approach doesn't work for categorical arrays. I've also realized something like `maximum(categorical([1 2; 3 4]), dims=1)` currently doesn't work for the same reason. Not sure what to do about this in general.